### PR TITLE
svgpath: General rework

### PR DESCRIPTION
**Pull request by peter-b**
- Don't handle any mouse events
- Use standard properties/property names
- Properly integrate with Property Browser
- More comprehensive documentation comments

[Rendered release notes](https://github.com/peter-b/livecode/blob/svgpath-rework/extensions/widgets/svgpath/notes/feature-rework.md)

### CODE REVIEW

File: extensions/widgets/svgpath/notes/feature-rework.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+* The SVG path widget's paints are now controlled by the standard LiveCode
+  **foregroundColor** and **hiliteColor** properties, and can use patterns.
+  They can be edited in the Property Inspector.
+
+# Properties
+
+* All properties are now fully integrated into the IDE Property Inspector.
+
+* The **iconColor** and **iconHiliteColor** properties have been removed.  Set
+  the standard LiveCode **foregroundColor** and **hilitoColor** properties
```
Comment (livecodeali) [inline extensions/widgets/svgpath/notes/feature-rework.md]: hilitoColor

File: extensions/widgets/svgpath/notes/feature-rework.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+* The **isHilited** property has been removed.  Use the standard LiveCode
+  **hilited** property instead.
+
+* The **iconPathPreset** property has been renamed to **iconPresetName** to
+  more accurately reflect the information it contains.
+
+* The **flipVertically** property has been renamed to **flipped**.
+
+* The **iconAngle** property has been renamed to **rotateAngle**.
```
Comment (livecodeali) [inline extensions/widgets/svgpath/notes/feature-rework.md]: should this just be `angle`, by analogy with the existing property for images?

File: extensions/widgets/svgpath/support/defaultscript.livecodescript [hunk 3]
```diff
@@ -0,0 +1,4 @@
+﻿script "com.livecode.widget.svgpath2.__DefaultScript"
```
Comment (livecodeali) [inline extensions/widgets/svgpath/support/defaultscript.livecodescript]: this still has svgpath2 in the name

File: extensions/widgets/svgpath/svgpath.lcb [hunk 4]
```diff
@@ -11,198 +31,190 @@ use com.livecode.library.widgetutils
...
-
-Example: set the iconHiliteColor of widget "SVG Icon" to "255,255,0,128"
+Summary: Determines the color of the icon when highlighted.
 
 Description:
-Use the iconHiliteColor property to set the color of the widget icon when clicked.
-The color is in the form of a 3 or 4 item string, the 4th optional item being the
-alpha value, all between 0 and 255.
-
+Use the iconHiliteColor property to set the color of the SVG path, when the
```
Comment (livecodeali) [inline extensions/widgets/svgpath/svgpath.lcb]: this needs changing to `hiliteColor`

File: extensions/widgets/svgpath/svgpath.lcb [hunk 5]
```diff
@@ -272,33 +269,15 @@ end handler
...
-  put "up" into mState
-	redraw all
-end handler
-
-public handler OnClick()
 	if the click count is 2 then
 		post "mouseDoubleUp" with [intToString(the click button)]
+	else
+		post "mouseUp" with [intToString(the click button)]
 	end if
```
Comment (livecodeali) [inline extensions/widgets/svgpath/svgpath.lcb:277]: should we just be using `the click button formatted as string` now?

**Comment by livecodeali** [comment]
@livecode-vulcan review ok dc861d3

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok dc861d3f5e2a591a9dcac433b1f7acf04bce620b

**Comment by livecode-vulcan** [comment]
😎 test success dc861d3f5e2a591a9dcac433b1f7acf04bce620b
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2015)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2028)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2009)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2022)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2031)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2037)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2042)

**Pull request merged**